### PR TITLE
Add feature 22

### PR DESCRIPTION
**Pull request by antonioCaballero17**
Adding JSON text editing package.
It is necessary refactor test because the method enterText for tester only works with TextField

### CODE REVIEW

**Comment by ashitaprasad** [comment]
Thank you for submitting the PR. I will review it soon.

File: pubspec.yaml [hunk 1]
```diff
@@ -46,10 +47,11 @@ dependencies:
   json_data_explorer:
     git:
       url: https://github.com/foss42/json_data_explorer.git
-    version: ^0.1.1
+      version: ^0.1.1
```
Comment (ashitaprasad) [inline pubspec.yaml:50]: Please revert this change as it was correct as per spec.

File: pubspec.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
+
```
Comment (ashitaprasad) [inline pubspec.yaml:13]: Please revert this change.

File: lib/widgets/editor.dart [hunk 3]
```diff
@@ -62,7 +68,22 @@ class _TextFieldEditorState extends State<TextFieldEditor> {
           insertTab();
         },
       },
-      child: TextFormField(
```
Comment (ashitaprasad) [inline lib/widgets/editor.dart:65]: Do not remove TextFormField. The code should switch between existing field & JsonTextField based on `widget.contentType == ContentType.json`

File: lib/widgets/editor.dart [hunk 4]
```diff
@@ -62,7 +68,22 @@ class _TextFieldEditorState extends State<TextFieldEditor> {
...
+      child: JsonTextField(
```
Comment (ashitaprasad) [inline lib/widgets/editor.dart:71]: When I switch from dark mode to light mode the theme of the entered JSON does not switch.

File: lib/widgets/editor.dart [hunk 5]
```diff
@@ -62,7 +68,22 @@ class _TextFieldEditorState extends State<TextFieldEditor> {
...
+        stringHighlightStyle: kCodeStyle.copyWith(
```
Comment (ashitaprasad) [inline lib/widgets/editor.dart:72]: The highlight style we follow are provided in the `consts.dart` - `kLightCodeTheme` & `kDarkCodeTheme`

File: lib/widgets/editor.dart [hunk 6]
```diff
@@ -62,7 +68,22 @@ class _TextFieldEditorState extends State<TextFieldEditor> {
...
+          color: Theme.of(context).colorScheme.secondary,
+        ),
+        keyHighlightStyle: kCodeStyle.copyWith(
+          color: Theme.of(context).colorScheme.primary,
+          fontWeight: FontWeight.bold,
+        ),
+        errorContainerDecoration: BoxDecoration(
```
Comment (ashitaprasad) [inline lib/widgets/editor.dart:79]: The error message should not be shown in the built-in container in the widget, but provided as a String callback just like `onChanged` providing the current `message` or null if no message. This message can then be state handled and displayed properly via a custom widget.

File: test/widgets/editor_test.dart [hunk 7]
```diff
@@ -26,11 +27,10 @@ void main() {
       ),
     );
```
Comment (ashitaprasad) [inline test/widgets/editor_test.dart:29]: Add new test cases when contenttype passed to TextFieldEditor is JSON.
Revert changes to these existing test cases.

**Comment by ashitaprasad** [comment]
Thanks for the PR @antonioCaballero17 
I have added my feedback above.

**Comment by ashitaprasad** [comment]
Hi @antonioCaballero17 
Any updates on the PR?

**Comment by antonioCaballero17** [comment]
give me a week, I am on holidays!.

**Comment by ashitaprasad** [comment]
Sure! No worries. 😌
Enjoy!

File: lib/widgets/editor.dart (hunk 6 above)
Comment (antonioCaballero17) [reply lib/widgets/editor.dart:79]: Sure, any UI/UX design or idea for error message?

Comment (ashitaprasad) [reply lib/widgets/editor.dart:79]: A small floating warning or error floating icon on the top-right which is visible whenever there is an error and shows the full error message when the user hovers over it (tooltip).

**Comment by ashitaprasad** [comment]
@antonioCaballero17 I have updated the PR to resolve conflicts, so please take a pull before you resume working on it.

**Comment by ashitaprasad** [comment]
Hi @antonioCaballero17 
Just wanted to check if there are any updates in regard to this feature.
Thanks!
